dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -91,6 +104,8 @@ Block WindowBlockInputStream::readImpl()
         tryCalculate();
     }
 
+    if (auto res = returnIfCancelledOrKilled(); res.has_value())
+        return res.value();
```

Comment:
When  query was canceled or killed, we should return {} or Block{}?

Reply:
https://github.com/pingcap/tiflash/blob/949bd03b1fafb1c38dcbe59dde2860254e8cebfe/dbms/src/DataStreams/IProfilingBlockInputStream.cpp#L60-L63
same as IProfilingBlockInputStream, return Block{} directly.